### DISCUSSION TITLE
[Question]:  How to Configure BaseUrl Parameter for OpenAI Requests via Relay Service?

### DISCUSSION THREAD

**Discussion by forwaard** [danny-avila/LibreChat#2878] [Q&A]
### What is your question?

Hi there,

This is an awesome project!  

I need to configure the BaseUrl parameter because all my AI projects send requests to OpenAI through a relay service, but I can't seem to find where to set this up. 

I noticed there's an ASSISTANTS_BASE_URL environment variable in the ASSISTANTS configuration. Is there something like 
OPENAI_BASE_URL or something similar that I can configure?

### More Details

like below in code

```
import openai

openai.api_key = '...'
openai.base_url = "https://..."  # <---- config a relay service domain

completion = openai.chat.completions.create(
    model="gpt-4",
    messages=[
        {
            "role": "user",
            "content": "...",
        },
    ],
)
print(completion.choices[0].message.content)```


### What is the main subject of your question?

Endpoints

### Screenshots

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by zionlee-hot** [comment]
https://www.librechat.ai/docs/configuration/dotenv
Read this. For example, if you want to set reverse proxy for OpenAI, you can code in the .env:  OPENAI_REVERSE_PROXY=whatever you want.

**Comment by Originalimoc** [comment]
https://www.librechat.ai/docs/configuration/librechat_yaml/ai_endpoints

**Comment by ogencoglu** [comment]
IS there a way to set the base_url in UI that us hosted in huggingface : https://librechat-librechat.hf.space/c/new
